src/ray/util/util.h

```diff
@@ -3,17 +3,27 @@
 
 #include <chrono>
 
-/// Return the number of milliseconds since the Unix epoch.
+/// Return the number of milliseconds since the steady clock epoch. NOTE: The
+/// returned timestamp may be used for accurately measuring intervals but has
+/// no relation to wall clock time. It must not be used for synchronization
+/// across multiple nodes.
 ///
 /// TODO(rkn): This function appears in multiple places. It should be
 /// deduplicated.
 ///
-/// \return The number of milliseconds since the Unix epoch.
-int64_t current_time_ms() {
+/// \return The number of milliseconds since the steady clock epoch.
+inline int64_t current_time_ms() {
```

Comment:
`current_time_ms` uses `std::chrono::steady_clock` and `current_sys_time_ms` uses `std::chrono::system_clock`. The former is monotonic but has no relation to wall clock time, whereas the latter has no monotonicity guarantees but is supposed to follow wall clock time as closely as possible.

Since I was using expiration timestamps for the leases, I added `current_sys_time_ms`. It's nice to have, but I guess not necessary once I implement the change mentioned above to switch the lease entry to contain the timeout period instead of the expiration timestamp.

Reply:
Ok, I see.